app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,40 @@
+class PostsController < ApplicationController
+    def create
+        @post = current_user.posts.build(post_params)
+        if @post.save
+            redirect_back(fallback_location: root_path)
+            flash[:notice] = "Post successfully created"
+        else
+            flash[:error] = @post.errors.full_messages
```

Comment:
wonderful work using the error provided by rails 💪 
how ever:
- redirect is a full request, we can just use render here
- the errors are also available to the form we can trap it there and us a partial to display it. that way errors can share that partial consider this [post](https://learn.co/lessons/validations-with-form_for-rails)
[OPTIONAL]